src/core/fieldformatter/qgsvaluerelationfieldformatter.cpp

```diff
@@ -142,3 +142,23 @@ QgsValueRelationFieldFormatter::ValueRelationCache QgsValueRelationFieldFormatte
 
   return cache;
 }
+
+QStringList QgsValueRelationFieldFormatter::valueToStringList( const QVariant &value )
```

Comment:
Would it make sense to return a `QVariantList` here? QVariants can always be converted to QStrings later on, while the opposite is not possible.

Reply:
@m-kuhn the values returned by the function were always used in a `QStringList` in the original code (and checked as strings).

Reply:
Because they were written at a time where we had no array support and only string parsing.